### ISSUE TITLE
[1.19-Forge] Java Heap Space crash after awhile.

### ISSUE THREAD

**Issue by Gbergz** [t2pellet/strawgolem#90]
**Game Info**
Game Version : 1.19
Mod Version : 2.0.0b1
Mod Loader : Forge

**Describe the bug**
After awhile, server lags and server crashes due to Java Heap Space.

**To Reproduce**
Not sure really how to, just have the server on, use some golems.
Profit.

**Screenshots**
No response.

**Additional context**
Crash report: https://gist.github.com/Gbergz/f434f23f838de384d6b065cff5923f7d

Edit : Maybe a cross issue with croptopia?

**Comment by t2pellet** [comment]
The old croptopia issue with saplings has been fixed, this appears to be something else most likely.

Out of curiosity, once the server crashes the first time, does it crash much sooner on subsequent launches?

I'm trying to see if the root of this is a bug with the OctTrees leaking memory somehow, or if its actually them just taking up too much space.

**Comment by Gbergz** [comment]
> Out of curiosity, once the server crashes the first time, does it crash much sooner on subsequent launches?

I'm not quite sure actually. Maybe.

**Comment by t2pellet** [comment]
@Gbergz are you able to download this build? 

https://github.com/t2pellet/strawgolem/suites/7015633357/artifacts/275533392

If so, try giving it a whirl and see if it helps

**Comment by t2pellet** [comment]
Try using the forge artifact from this run:

https://github.com/t2pellet/strawgolem/actions/runs/2532809336

(open the zip file and use the regular jar, not the dev or sources ones)

Let me know if that fixes the issue for ya?

**Comment by t2pellet** [comment]
@Gbergz  let me know if this is resolved with the build I sent!

**Comment by Gbergz** [comment]
Thing is the lag is no longer happening so I can't really figure out if the build fixed it or not. Haven't tried it yet though, haven't had time.

**Comment by Gbergz** [comment]
Still happens on that version.
Server went on to hang for 720s (which is the server watchdog timer).

**Crash-log:**
https://gist.github.com/Gbergz/19d268633a19e24decec3d07a41867c2

**Comment by t2pellet** [comment]
And if you remove strawgolem entirely the problem goes away?

**Comment by Gbergz** [comment]
Yeah, that. And if I edit the config below.
```
#Golem lifespan in ticks. Set to -1 for infinite
lifespan = 168000 to 1
```

Setting it to 1 solves it, server no longer dies.

**Comment by Gbergz** [comment]
So far, so good with that build. Been testing for a bit now.

**Comment by Gbergz** [comment]
Nevermind exactly when I posted that, the server crashed hard.
might just be a new issue entirely
https://gist.github.com/Gbergz/6b487386794b70b36d74af06f69893d2

**Comment by t2pellet** [comment]
@Gbergz that does look like a separate issue, weird one too, looks like it was getting a null somewhere comparing candidate nearest crops.

Not quite sure why that would happen, but the build below *should* protect against that, or at least provide better logging
https://github.com/t2pellet/strawgolem/actions/runs/2578485081

**Comment by t2pellet** [comment]
@Aiyrie this build should fix that issue:
https://github.com/t2pellet/strawgolem/runs/7099464455

**Comment by Aiyrie** [comment]
> @Aiyrie this build (when completed) should fix that issue: https://github.com/t2pellet/strawgolem/actions/runs/2578542473

Sweet! Thank you :D 

After enough time testing it on the server, the out of ram error is fixed!
I've ran into a new error however with a crash log and auto server reset with the new batch lol

[Context of what occurred seconds before crash] 
Two Straw Golem's are fenced in with 1 chest & some crops [Mod: Croptopia 2.0.5] with "Peach Sapling"'s planted around the area containing the Straw Golem's.

Here's my server's crash log, I hope it helps!
[Crash.txt](https://github.com/t2pellet/strawgolem/files/9006226/Crash.txt)

**Comment by Gbergz** [comment]
Keep getting the out of memory crashes, still.
Haven't tried the build above where Ayrie. But will do whenever I got time. Will that fix it? That needs to be on CurseForge if so.

**Comment by t2pellet** [comment]
Should be fixed in 2.0.0.beta-3
